src/LeanCode.ContractsGenerator/ContractsCompiler.cs

```diff
@@ -1,51 +1,59 @@
-
-        private static readonly List<MetadataReference> DefaultAssemblies = new()
+        private static readonly ImmutableList<string> ImplicitGlobalUsings = ImmutableList.CreateRange(new[]
         {
-            MetadataReference.CreateFromFile(typeof(IQuery<>).Assembly.Location),
-            MetadataReference.CreateFromFile(typeof(ICustomAuthorizer<>).Assembly.Location),
-            MetadataReference.CreateFromFile(typeof(List<>).Assembly.Location),
-            MetadataReference.CreateFromFile(typeof(Date).Assembly.Location),
-            MetadataReference.CreateFromFile(typeof(Uri).Assembly.Location),
-            MetadataReference.CreateFromFile(typeof(HttpClient).Assembly.Location),
-            MetadataReference.CreateFromFile(Path.Combine(ObjectAssemblyPath, "mscorlib.dll")),
-            MetadataReference.CreateFromFile(Path.Combine(ObjectAssemblyPath, "System.dll")),
-            MetadataReference.CreateFromFile(Path.Combine(ObjectAssemblyPath, "System.Core.dll")),
-            MetadataReference.CreateFromFile(Path.Combine(ObjectAssemblyPath, "System.Runtime.dll")),
-            MetadataReference.CreateFromFile(Path.Combine(ObjectAssemblyPath, "System.Private.Uri.dll")),
-        };
-
-        static ContractsCompiler()
+            "System",
+            "System.Collections.Generic",
+            "System.IO",
+            "System.Linq",
+            "System.Net.Http",
+            "System.Threading",
+            "System.Threading.Tasks",
+        });
+
+        public static readonly ImmutableList<string> ReferenceAssemblyNames = ImmutableList.CreateRange(new[]
         {
-            if (string.IsNullOrEmpty(ObjectAssemblyPath))
-            {
-                throw new InvalidProjectException("Cannot locate `System.Runtime.dll`. Please check if the generator is run as a framework-dependent executable.");
-            }
-        }
+            "System.Collections.Reference",
+            "System.Linq.Reference",
+            "System.Net.Http.Reference",
+            "System.Runtime.Reference",
+            "System.Runtime.Extensions.Reference",
+        });
+
+        public static readonly ImmutableList<string> LeanCodeAssemblyNames = ImmutableList.CreateRange(new[]
+        {
+            "LeanCode.CQRS",
+            "LeanCode.CQRS.Security",
+            "LeanCode.Time",
+        });
+
+        public static readonly ImmutableList<PortableExecutableReference> ReferenceAssemblies = DependencyContext.Default.CompileLibraries
+            .Where(cl => cl.Type == "referenceassembly")
+            .Where(cl => ReferenceAssemblyNames.Contains(cl.Name, StringComparer.InvariantCultureIgnoreCase))
+            .SelectMany(cl => cl.ResolveReferencePaths())
+            .Select(path => MetadataReference.CreateFromFile(path))
+            .ToImmutableList();
+
+        public static readonly ImmutableList<PortableExecutableReference> DefaultAssemblies = ReferenceAssemblies
```

Comment:
I would say - let's make the concatenation on `names` first, then resolve all the `MetadataReference`s in a single expression.

Reply:
No concatenation of names, just checking two separate sets, but all references are now resolved at once.